### PR TITLE
feat: detect safari 10.1+ in IDB adapter

### PR DESCRIPTION
**Pull request by janl**
PR for https://github.com/pouchdb/pouchdb/pull/6173

### CODE REVIEW

File: packages/node_modules/pouchdb-adapter-idb/src/index.js [hunk 1]
```diff
@@ -807,6 +807,15 @@ IdbPouch.valid = function () {
   var hasFetch = typeof fetch === 'function' &&
     fetch.toString().indexOf('[native code') !== -1;
 
+  // Safari > 10.1 has a good IDB implementation
+  // this check is valid until Safari version 19.99
```
Comment (mgol) [inline packages/node_modules/pouchdb-adapter-idb/src/index.js]: 19.99? It only checks for some 10.x & 11.x versions.

File: packages/node_modules/pouchdb-adapter-idb/src/index.js [hunk 2]
```diff
@@ -807,6 +807,15 @@ IdbPouch.valid = function () {
...
+  // at which point I hope this is all refactoreed anyway
+  var isSafariWithGoodIDB = isSafari && (
+    /Version\/10.[1-9]{1,2}/.test(navigator.userAgent)
+    ||
+    /Version\/11.[0-9]{1,2}/.test(navigator.userAgent)
```
Comment (mgol) [inline packages/node_modules/pouchdb-adapter-idb/src/index.js]: Shouldn't it be `/Version\/1[1-9].[0-9]{1,2}/`?

Comment (janl) [inline packages/node_modules/pouchdb-adapter-idb/src/index.js]: you’re right, good catch :)

fixing in next push

**Comment by nolanlawson** [comment]
I feel my original PR (https://github.com/pouchdb/pouchdb/pull/6173) may be doing a better job of feature detection… Instead of UA-sniffing, I'm sniffing for `fetch()` which happens to coincide with the release of a decent IDB engine in Safari. This will also work for browsers based on WebKit that for whatever reason might not match the UA string. Feels safer to me.

**Comment by janl** [comment]
Oh, then I misunderstood the issue you were raising.

**Comment by janl** [comment]
@nolanlawson I thought you were asking for an addition to your PR about the version number (just because the build status is still red on the PR). If that’s wrong, what are the next steps here?

**Comment by nolanlawson** [comment]
@janl I just looked at the build output for the most recent job, and the Safari UA is `10.0.1`. So unfortunately it seems SauceLabs has not updated Safari yet.

**Comment by nolanlawson** [comment]
I'd prefer to wait for https://github.com/pouchdb/pouchdb/pull/6173 to pass although at this point we've waited so long for SauceLabs to add Safari 10.3+ support that maybe we should just test manually and then cross our fingers.

**Comment by mgol** [comment]
@nolanlawson BrowserStack has Safari 10.1 & iOS 10.3 if that changes anything for you.

**Pull request closed without merging**